crates/parcel_plugin_resolver/src/resolver.rs

```diff
@@ -1,19 +1,328 @@
+    resolver
+      .conditions
+      .set(ExportsCondition::WORKER, dep.env.context.is_worker());
+    resolver.conditions.set(
+      ExportsCondition::WORKLET,
+      dep.env.context == EnvironmentContext::Worklet,
+    );
+    resolver
+      .conditions
+      .set(ExportsCondition::ELECTRON, dep.env.context.is_electron());
+    resolver
+      .conditions
+      .set(ExportsCondition::NODE, dep.env.context.is_node());
+    resolver.conditions.set(
+      ExportsCondition::PRODUCTION,
+      self.mode == BuildMode::Production,
+    );
+    resolver.conditions.set(
+      ExportsCondition::DEVELOPMENT,
+      self.mode == BuildMode::Development,
+    );
+
+    resolver.entries = Fields::MAIN | Fields::MODULE | Fields::SOURCE;
+    if dep.env.context.is_browser() {
+      resolver.entries |= Fields::BROWSER;
+    }
+
+    resolver.include_node_modules = Cow::Borrowed(&dep.env.include_node_modules);
+
+    let resolver_options = ResolveOptions {
+      conditions: dep.package_conditions,
+      custom_conditions: vec![],
+      // TODO: Do we need custom condition?
+      // custom_conditions: dep.custom_package_conditions.clone(),
+    };
+
+    let resolve_from = dep
+      .resolve_from
+      .as_ref()
+      .or(dep.source_path.as_ref())
+      .as_ref()
+      .map(|p| Cow::Borrowed(p.as_path()))
+      .unwrap_or_else(|| Cow::Owned(self.project_root.join("index")));
+
+    let mut res = resolver.resolve_with_options(
+      specifier,
+      &resolve_from,
+      match dep.specifier_type {
+        SpecifierType::CommonJS => parcel_resolver::SpecifierType::Cjs,
+        SpecifierType::Esm => parcel_resolver::SpecifierType::Esm,
+        SpecifierType::Url => parcel_resolver::SpecifierType::Url,
+        // TODO: what should specifier custom map to?
+        SpecifierType::Custom => parcel_resolver::SpecifierType::Esm,
+      },
+      resolver_options,
+    );
+
+    let side_effects = if let Ok((parcel_resolver::Resolution::Path(p), _)) = &res.result {
+      match resolver.resolve_side_effects(p, &res.invalidations) {
+        Ok(side_effects) => side_effects,
+        Err(err) => {
+          res.result = Err(err);
+          true
+        }
+      }
+    } else {
+      true
+    };
+
+    // TODO: Create diagnostics from errors
+    // TODO: Handle invalidations
+
+    match res.result? {
+      (parcel_resolver::Resolution::Path(path), _invalidations) => Ok(Resolution {
+        file_path: path,
+        side_effects,
+        can_defer: false,
+        code: None,
+        is_excluded: false,
+        meta: JSONObject::default(),
+        pipeline: None,
+        priority: None,
+        query: None,
+      }),
+      (parcel_resolver::Resolution::Builtin(builtin), _invalidations) => {
+        self.resolve_builtin(ctx, builtin)
+      }
+      (parcel_resolver::Resolution::Empty, _invalidations) => Ok(Resolution {
+        file_path: self
+          .project_root
```

Comment:
is project_root the parcel root?

Reply:
No, I should have put a todo here as well. This is a temporary hack, we'll need a better way to resolve the empty file.